src/error.jl

```diff
@@ -1,14 +1,43 @@
+"""
+    MPIError
+
+Error thrown when an MPI function returns an error code. The `code` field contains the MPI error code.
+"""
 struct MPIError <: Exception
     code::Cint
 end
+function Base.show(io::IO, err::MPIError)
+    print(io, "MPIError(", err.code, "): ", error_string(err))
+end
+
+"""
+    FeatureLevelError
+
+Error thrown if a feature is not implemented in the current MPI backend.
+"""
+struct FeatureLevelError <: Exception
+    function_name::Symbol
+    min_version::VersionNumber # minimal MPI version required for this feature to be available
+end
+function Base.show(io::IO, err::FeatureLevelError)
+    print(io, "FeatureLevelError($(err.function_name)): Minimum MPI version is $(err.min_version)")
+end
+
+macro mpichk(expr, min_version=nothing)
+    if !isnothing(min_version) && expr.args[2].head == :tuple
+        fn = expr.args[2].args[1].value
+        if dlsym(Libdl.dlopen(libmpi, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL), fn; throw_error=false) == C_NULL
```

Comment:
```suggestion
        if isnothing(dlsym(Libdl.dlopen(libmpi, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL), fn; throw_error=false))
```

Reply:
Maybe we should keep the libmpi handle around, instead of running `dlopen` every time?  We already have it for JLL packages.